### PR TITLE
Adding overload RegisterCollection<TService>(params Type[] serviceTypes)

### PR DESCRIPTION
**Pull request by abatishchev**

### CODE REVIEW

**Comment by dotnetjunkie** [comment]
Thanks for this pull request. I absolutely do understand the reason for this pull request, but there is actually a good reason why this overload does not exist. It is done for consistency of the API. The omission of this overload is a conscious design decision. 

If we add this particular overload, users would expect the following overload as well:

``` c#
public void RegisterCollection(Type serviceType, params Type[] serviceTypes);
```

Having one without the other is confusing. This particular non-generic overload however, can't be added, because the C# compiler will never call it. Example:

``` c#
container.RegisterCollection(typeof(IX), typeof(A), typeof(B));
```

Here is seems that we are calling the previously defined overload, but in fact, what the C# compiler will generate is the following call:

``` c#
container.RegisterCollection<Type>(new[] { typeof(IX), typeof(A), typeof(B) });
```

In other words, the `Type` type is used as service, which is certainly not what the developer expected here.

There are other places in the API where we deliberately left out the `params` overload, because of consistency.

**Comment by abatishchev** [comment]
I see, makes sense.

As a workaround, I use `container.RegisterCollection<Type>(new[] { typeof(Service) })`.

**Comment by dotnetjunkie** [comment]
Yes, this is exactly the intended use. It's not the greatest syntax, but having an inconsistent syntax is even worse. 

In the common use case however, you will see that `RegisterCollection` will be used with a list of types that is built elsewhere and in that case, you will not have to wrap the list of types with `new[] { ... }`. For instance:

``` c#
container.RegisterCollection<IService>(
    from type in container.GetTypesToRegister(typeof(IService), applicationAssemblies)
    where type != typeof(SpecialService)
    select type);
```
